### ISSUE TITLE
Carthage Testing: Curators

### ISSUE THREAD

**Issue by dmtrjsg** [Joystream/pioneer#3411]
Playground details are covered in this issue: 

* https://github.com/Joystream/pioneer/issues/3217

:warning: Content WG lead needs to be hired before this can be executed

### Resources File

[testing-resources.zip](https://github.com/Joystream/pioneer/files/9222719/testing-resources.zip)

### Curators Test Steps
* [ ] Content Lead to create 15 video
  * See resources
* [ ] Hire (at least) 3 more curators
* [ ] Create multiple curator groups, with multiple permissions.
  * [ ] Try all actions for each group, eg. delete, edit, censor channel and video, and verify what should be allowed is, and vice versa.
  * [ ] Change the permissions for a group or two, and verify that the new permissions are in effect
  * [ ] Check if the Lead can perform all actions without being in any group

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1202132419573087/1202685466093307) by [Unito](https://www.unito.io)

**Comment by freakstatic** [comment]
Video categories blocked by this
https://github.com/Joystream/joystream/issues/4119

**Comment by freakstatic** [comment]
Channel categories not supported in the CLI? :thinking: 

```
 ›   Warning: content:createChannelCategory is not a joystream-cli command.
Did you mean content:createVideoCategory? [y/n]: 
 ›   Error: Run joystream-cli help content for a list of available commands.
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by dmtrjsg** [comment]
@freakstatic we don’t have channelCategories in Carthage since they are not being used.

**Comment by dmtrjsg** [comment]
Since Curators will no longer be responsible for categories, this ticket does not need to be tested. Closing as such 

cc 
@kdembler @freakstatic @singulart

**Comment by kdembler** [comment]
@dmtrjsg It seems there's more to this ticket than just categories though?

**Comment by dmtrjsg** [comment]
@kdembler yes you are right 🙈, that was a premature move, I take it back. Thank you for keeping this in check 🙇  👍

**Comment by dmtrjsg** [comment]
@freakstatic following the sync call we've got the update, and similarly to channel permissions, the curators permissions in Carthage can only be set up, but not updated. Closing this ticket in the meantime, and will link a new issue to it, once we are able to test it fully.  I am moving this to the mainnet testing scope, and moving to out of scope for the current round.


Could you perhaps update if this ticket was partially testable, so we can better define the scope of focus for the follow up rounds?

**Comment by freakstatic** [comment]
I was able to test the following:
## [x] Hire (at least) 3 more curators:

Applications: [#465678](https://polkadot.js.org/apps/?rpc=wss://54.172.154.45.nip.io/ws-rpc#/explorer/query/0x878b43127f459bd2b46dea34f9efefb0146b1116eea991e09d3f592384764eaf), [#465663](https://polkadot.js.org/apps/?rpc=wss://54.172.154.45.nip.io/ws-rpc#/explorer/query/0x7ec5f311d5e1c2bb4975116b7b6a64011bb9a1adf1508c76f06328b24c2e4685)
[#460457](https://polkadot.js.org/apps/?rpc=wss://54.172.154.45.nip.io/ws-rpc#/explorer/query/0x70082e1d338229f2508cbdfb59c23645ef804991b4183f1f7595c5d37c4c2e74)
Hire on [465,713](https://polkadot.js.org/apps/?rpc=wss://54.172.154.45.nip.io/ws-rpc#/explorer/query/0x11514e7044878aa143255805b1a396a976d1effa8fdd7b4ac2e225a93c96d283),
No problems were found.

I wasn't able to test:
## [] Content Lead to create 15 video
- Storage providers and policies were not set
## [] Create multiple curator groups, with multiple permissions.
- Found some problems with permissions examples:
     `"HideVideo": true` is not accepted as a valid permission

**Comment by dmtrjsg** [comment]
@freakstatic thanks a mil for your update, which network did you test this on?

**Comment by freakstatic** [comment]
> @freakstatic thanks a mil for your update, which network did you test this on?

It was on wss://54.172.154.45.nip.io/ws-rpc 
@kdembler suggested to use `wss://atlas-next.joystream.org/ws-rpc` but didn't got the chance to do it. 
It's worthy to do it now?

**Comment by dmtrjsg** [comment]
@freakstatic yep, also got this one in the bottom from Klaudiusz. I'd say its worth taking for a spin! Thanks for asking, reopening.

**Comment by freakstatic** [comment]
# WIP


Network: wss://atlas-next.joystream.org/ws-rpc

## Check if the Lead can perform all actions without being in any group

### Lead Channel Actions

Trying to delete a channel without being a owner fails which I think it was expected but I'm not sure why it exists the "curator" context :thinking: 
  `yarn run joystream-cli content:deleteChannel -c 18` or `yarn run joystream-cli content:deleteChannel -c 18 --context=Curator`
<details>
  <summary>Log</summary>

```
Error: No account found with access to manage the provided channel
error Command failed with exit code 45.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
 ```
  </details>
<br>

Deleting a channel as the curator lead without videos works fine. [#107,326](https://polkadot.js.org/apps/?rpc=wss://atlas-next.joystream.org/ws-rpc#/explorer/query/0x69f9c87ad310e03e40116a06f58649738854e9576f5e2ce41f7dacd0c16b5401)
  `yarn run joystream-cli content:deleteChannelAsModerator -c 18 -r "testing"`
  <details>
    <summary>Log</summary>

```
    > Derived context: Lead
    ? Are you sure you want to remove channel 18? Yes
    > Sending content.deleteChannelAsModerator extrinsic from bob...
    Tx params: [ [ 'Lead', '18', '0', 'testing' ] ]
    ? Tx fee of 61.5352 mJOY will be deduced from you account, do you confirm the transfer? Yes
    Extrinsic successful!
```
  </details>
<br>


Trying to delete a channel with associated videos fails, as expected I think?
`yarn run joystream-cli content:deleteChannelAsModerator -c 17 -r "testing"`
<details>
<summary>Log</summary>

```
> Derived context: Lead
 ›   Error: This channel still has 2 associated video(s)!
 ›   Delete the videos first using content:deleteVideo command
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
</details>

but `--force` doesn't remove the videos associated with the channel, was it expected? @thesan 
`yarn run joystream-cli content:deleteChannelAsModerator -c 17 -r "testing" --force`
<details>
<summary>Log</summary>

```
> Derived context: Lead
 ›   Error: This channel still has 2 associated video(s)!
 ›   Delete the videos first using content:deleteVideo command
error Command failed with exit code 2.
```
</details>
<br>


### Lead Video Actions
Trying to delete a video without being the owner fails as expected but I'm not sure why it exists the "curator" context here as well :thinking: 
<details>
<summary>Log</summary>

```
`yarn run joystream-cli content:deleteVideo -v 22 --context=Curator`
>  ›   Error: No account found with access to manage the provided channel
error Command failed with exit code 45.
```
</details>
<br>

Trying to delete a video with associated data objects fails, but it will not always have some type of data object associated? So the `--force` parameter will be always required.
`yarn run joystream-cli content:deleteVideoAsModerator -v 22 -r "testing"`
<details>
<summary>Log</summary>

```
> Derived context: Lead
Following data objects are still associated with the video:
75 - DataObjectTypeVideoMedia
76 - DataObjectTypeVideoThumbnail
 ›   Error: Cannot remove associated data objects unless --force flag is used
</details>
<br>

Trying to delete an video with a Nft associated fails was expected
`yarn run joystream-cli content:deleteVideoAsModerator -v 22 -r "testing" --force`
<details>
<summary>Log</summary>
```

> Derived context: Lead
Following data objects are still associated with the video:
75 - DataObjectTypeVideoMedia
76 - DataObjectTypeVideoThumbnail
Video state bloat bond of 333.3333 mJOY will be transferred to j4Uomosoo3JCKEjr677nLSUPsnDVMsEBbbXyqoUwyCu8bZKrh
Data objects state bloat bond of 3.3333 mJOY will be repaid with accordance to the bloat bond policy.
? Are you sure you want to remove video 22 and all associated data objects? Yes
Sending content.deleteVideoAsModerator extrinsic from bob...
Tx params: [ [ 'Lead', '22', '2', 'testing' ] ]
? Tx fee of 61.5330 mJOY will be deduced from you account, do you confirm the transfer? Yes
 ›   Error: Extrinsic failed! Extrinsic execution error: NftAlreadyExists (Nft for given video id already exists)
error Command failed with exit code 52.
</details>
<br>

**Comment by freakstatic** [comment]
Is still not possible to test the following points since permissions were moved to mainnet context
- [] Try all actions for each group, eg. delete, edit, censor channel and video, and verify what should be allowed is, and vice versa.

- [] Change the permissions for a group or two, and verify that the new permissions are in effect

**Comment by dmtrjsg** [comment]
Thanks @freakstatic for pushing it to the realms of possibilities of testing on the curtent version. For now suggesting to close this as completed as we will still need to test curators for Mainnet release, also inclusive some of the new runtime changes for this topic. 

Same comment [added to discord](https://discord.com/channels/811216481340751934/812344874099277824/1020334020552306768)
